src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -348,20 +347,33 @@ private JObject GetVerificationResult(UInt160 scriptHash, ContractParameter[] ar
             {
                 throw new RpcException(-100, "Unknown contract");
             }
-            var methodName = "verify";
-
-            Transaction tx = signers == null ? null : new Transaction
+            var md = contract.Manifest.Abi.GetMethod("verify", -1);
+            if (md is null)
+                throw new RpcException(-101, $"The smart contract {contract.Hash} haven't got verify method.");
+            if (md.ReturnType != ContractParameterType.Boolean)
+                throw new RpcException(-102, "The verify method doesn't return boolean value.");
+
+            var defaultSigner = new List<Signer>() { new() { Account = scriptHash } };
+            if (signers != null)
             {
-                Signers = signers.GetSigners(),
-                Attributes = Array.Empty<TransactionAttribute>()
+                defaultSigner.AddRange(signers.GetSigners());
```

Comment:
Can we make here `defaultSigner = signers.GetSigners();`?
It shouldn't be mandatory for the contract signer to be a sender. And if the user wants to provide signers by himself, then he should include the contract signer with the desired scope in the parameters list.

The other option is to check that `signers` does not contain the contract signer and if the contract signer is missing, we should add it to the `signers` list.

Reply:
Updated it.